WebKitBrowser/InjectedBundle/main.cpp

```diff
@@ -125,9 +146,54 @@ static class PluginHost {
         }
     }
 
+#ifdef UPDATE_TZ_FROM_FILE
+    static void HandleTimeZoneFileUpdate(GFileMonitor *monitor, GFile *file, GFile *other, GFileMonitorEvent evtype, gpointer user_data) {
+        if (evtype == G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT) {
+            gchar *content = nullptr;
+            gsize length = 0;
+
+            g_file_load_contents(file, nullptr, &content, &length, nullptr, nullptr);
+            if(content && length > 0) {
+                gchar *timezone = g_strdup_printf(":%s", content);
+
+                SYSLOG(Trace::Information, (_T("TimeZone is updated to %s from %s file"), timezone, kTimeZoneFile));
+                Core::SystemInfo::SetEnvironment(_T("TZ"), _T(timezone), true);
+                tzset();
+
+                g_free(timezone);
+                g_free(content);
+            }
+
+            if(user_data && monitor) {
+                gulong *handlerId = (gulong*)user_data;
+                g_signal_handler_disconnect(monitor, *handlerId);
+                *handlerId = 0;
+            }
+        }
+    }
+
+    void UpdateTimeZone() {
+        if(kTimeZoneFile == nullptr || strlen(kTimeZoneFile) == 0) {
+            SYSLOG(Trace::Warning, (_T("Invalid file input for TZ update")));
+            return;
+        }
+
+        GFile *file = g_file_new_for_path(kTimeZoneFile);
+        if(g_file_query_exists(file, nullptr)) {
+            HandleTimeZoneFileUpdate(nullptr, file, nullptr, G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT, nullptr);
+        } else {
+            _timeZoneFileMonitor = g_file_monitor_file(file, G_FILE_MONITOR_NONE, nullptr, nullptr);
+            _timeZoneFileMonitorId = g_signal_connect(_timeZoneFileMonitor, "changed", reinterpret_cast<GCallback>(HandleTimeZoneFileUpdate), &_timeZoneFileMonitorId);
+            SYSLOG(Trace::Information, (_T("Installed file monitor for %s"), kTimeZoneFile));
```

Comment:
Timezone might be updated after the browser startup. So, I think it makes sense to keep the monitor running

Reply:
Updated implementation.